.github/workflows/release.yml

```
@@ -58,115 +106,105 @@ jobs:
+        run: |
+          echo "$DEBIAN_PRIVATE_KEY" | base64 --decode | gpg --batch --import
+          gpg --list-secret-keys --keyid-format LONG
+      - name: Build rpm repository
         env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          CODENAME: stable
+          DEBIAN_PRIVATE_KEY: ${{ secrets.DEBIAN_PRIVATE_KEY }}
+        run: ./build rpm-repo --configuration=Release --exclusive
+        if: ${{ env.DEBIAN_PRIVATE_KEY }}
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v4
         with:
-          upload_url: ${{ steps.release_data.outputs.upload_url }}
-          asset_path: _build/deb/ckan_${{ steps.release_data.outputs.deb_version }}_all.deb
-          asset_name: ckan_${{ steps.release_data.outputs.version }}_all.deb
-          asset_content_type: application/vnd.debian.binary-package
-      - name: Upload ckan-*.rpm
-        uses: actions/upload-release-asset@v1.0.1
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          aws-region: us-east-1
+      - name: Push rpm to S3
+        run: aws s3 sync _build/rpm/repo s3://${AWS_S3_BUCKET}/rpm/stable --follow-symlinks
+      - name: Upload RPM release
+        run: gh release upload ${{ github.event.release.tag_name }} _build/rpm/RPMS/noarch/ckan-${RPM_VERSION}-1.noarch.rpm
         env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+  upload-binaries:
+    runs-on: ubuntu-latest
+    needs:
+      - test-release
+      - smoke-inflator
+    env:
+      GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+    steps:
+      - uses: actions/download-artifact@v4
         with:
-          upload_url: ${{ steps.release_data.outputs.upload_url }}
-          asset_path: _build/rpm/RPMS/noarch/ckan-${{ steps.release_data.outputs.rpm_version }}-1.noarch.rpm
-          asset_name: ckan-${{ steps.release_data.outputs.version }}-1.noarch.rpm
-          asset_content_type: application/x-rpm
+          name: Release-repack-unsigned
+          path: _build/repack/
+      - run: ls -lah _build/repack/
+      - name: Upload ckan.exe
+        run: gh release upload ${{ github.event.release.tag_name }} _build/repack/Release/ckan.exe
       - name: Upload AutoUpdater.exe
-        uses: actions/upload-release-asset@v1.0.1
-        env:
-          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-        with:
-          upload_url: ${{ steps.release_data.outputs.upload_url }}
-          asset_path: _build/repack/Release/AutoUpdater.exe
-          asset_name: AutoUpdater.exe
-          asset_content_type: application/vnd.microsoft.portable-executable
+        run: gh release upload ${{ github.event.release.tag_name }} _build/repack/Release/AutoUpdater.exe
 
+  notify-discord:
+    runs-on: ubuntu-latest
+    needs:
+      - build-dmg
+      - build-deb
+      - build-rpm
+      - upload-binaries
```

```suggestion
    needs:
      - build-dmg
      - build-deb
      - build-rpm
      - upload-binaries
      - upload-nuget
```

Should `upload-nuget` be listed here so we get notified if it fails?